clients/ops-l1/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM ethereum/client-go:v1.10.16 AS base
+
+RUN apk add --no-cache jq yarn git linux-headers gcompat
+RUN /usr/local/bin/geth console --exec 'console.log(admin.nodeInfo.name)' --maxpeers=0 --nodiscover --dev 2>/dev/null | head -1 > /version.txt
+ENV GLIBC_KEY=https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub
+ENV GLIBC_KEY_FILE=/etc/apk/keys/sgerrand.rsa.pub
+ENV GLIBC_RELEASE=https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.35-r0/glibc-2.35-r0.apk
+RUN wget -q -O ${GLIBC_KEY_FILE} ${GLIBC_KEY} \
+    && wget -O glibc.apk ${GLIBC_RELEASE} \
+    && apk add glibc.apk --force
+
+# build forge
+from alpine as build-environment
+WORKDIR /opt
+RUN apk add clang lld curl build-base linux-headers git \
+    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh \
+    && chmod +x ./rustup.sh \
+    && ./rustup.sh -y
+RUN git clone https://github.com/foundry-rs/foundry /opt/foundry
+WORKDIR /opt/foundry
+RUN ls /opt/foundry
+RUN source $HOME/.profile && cargo build --release \
+    && strip /opt/foundry/target/release/forge \
+    && strip /opt/foundry/target/release/cast
+
+FROM base
+COPY --from=build-environment /opt/foundry/target/release/forge /usr/local/bin/forge
+COPY --from=build-environment /opt/foundry/target/release/cast /usr/local/bin/cast
+# end forge
+
+# Regenerate the L1 genesis file
+RUN mkdir /hive
+ADD genesis.json /genesis.json
+RUN date +%s | xargs printf "0x%x" > /hive/genesis_timestamp
+RUN GENESIS_TIMESTAMP=$(cat /hive/genesis_timestamp); jq ". | .timestamp = \"$GENESIS_TIMESTAMP\" " < ./genesis.json > /hive/genesis-l1.json
+
+# Inject the startup script
+ADD entrypoint.sh /entrypoint.sh
+RUN chmod +x /entrypoint.sh
+
+# Deploy contracts using Hardhat
+RUN git clone https://github.com/ethereum-optimism/optimistic-specs /opt
+RUN cd /opt && git submodule init && git submodule update
+RUN cp -r /opt/packages/contracts /hive/contracts
+# Defer until L1 rpc server is up; called from devnet.go
+ADD deploy.sh /hive-bin/deploy.sh
+RUN chmod +x /hive-bin/deploy.sh
+
+# Helper scripts; called from devnet.go
+ADD cat.sh /hive-bin/cat.sh
```

Comment:
This util isn't necessary - you can call `cat` directly.

Reply:
Commented below, but only executables in `hive-bin` are allowed to executed from the hive `Client`.